### PR TITLE
(manke) rook-ceph operator

### PR DESCRIPTION
**Pull request by cbarria**
Follow script will install rook-ceph v1.10.10, ceph v16.2.10 and cephcsi v3.7.2.
2 disks per node - replica 3 - mngr 2 - mon 3 and SSL false (ingress in HTTP).
nodes clean up script also included.

### CODE REVIEW

File: manke/rook-ceph/rook-ceph.sh [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+ceph mgr module enable rook
+ceph mgr module enable nfs
+ceph orch set backend rook
+
+# --- customize below this line ---
+
+kubectl apply -f cephblockpool.yaml
+kubectl apply -f ceph-storageclass.yaml
+kubectl patch storageclass rook-ceph-block -p '{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}'
+
```
Comment (jhoblitt) [inline manke/rook-ceph/rook-ceph.sh:125]: manke will need the same nfs exports as pillan.

File: manke/rook-ceph/scripts/rook-ceph-node-cleanup.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+set -x
+
+DEVS=(
+  nvme0n1
+  nvme2n1
+)
```
Comment (jhoblitt) [inline manke/rook-ceph/scripts/rook-ceph-node-cleanup.sh:8]: I'd prefer we don't commit this script if the cleanup is dangerous.  I started hacking something together last week but it isn't ready.  This is a generated script:

```bash
ssh rke@manke01 sudo rm -rf /var/lib/rook
ssh rke@manke01 'ls /dev/mapper/ceph-* | xargs -I% -- echo /sbin/dmsetup remove %'
ssh rke@manke01 sudo rm -rf /dev/ceph-*
ssh rke@manke01 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301083"
ssh rke@manke01 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301083" bs=1M count=100 oflag=direct,dsync
ssh rke@manke01 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301083"
ssh rke@manke01 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301316"
ssh rke@manke01 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301316" bs=1M count=100 oflag=direct,dsync
ssh rke@manke01 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301316"
ssh rke@manke01 lsblk
ssh rke@manke01 sudo reboot
ssh rke@manke02 sudo rm -rf /var/lib/rook
ssh rke@manke02 'ls /dev/mapper/ceph-* | xargs -I% -- echo /sbin/dmsetup remove %'
ssh rke@manke02 sudo rm -rf /dev/ceph-*
ssh rke@manke02 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300851"
ssh rke@manke02 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300851" bs=1M count=100 oflag=direct,dsync
ssh rke@manke02 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300851"
ssh rke@manke02 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301298"
ssh rke@manke02 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301298" bs=1M count=100 oflag=direct,dsync
ssh rke@manke02 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301298"
ssh rke@manke02 lsblk
ssh rke@manke02 sudo reboot
ssh rke@manke03 sudo rm -rf /var/lib/rook
ssh rke@manke03 'ls /dev/mapper/ceph-* | xargs -I% -- echo /sbin/dmsetup remove %'
ssh rke@manke03 sudo rm -rf /dev/ceph-*
ssh rke@manke03 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301082"
ssh rke@manke03 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301082" bs=1M count=100 oflag=direct,dsync
ssh rke@manke03 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301082"
ssh rke@manke03 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300798"
ssh rke@manke03 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300798" bs=1M count=100 oflag=direct,dsync
ssh rke@manke03 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300798"
ssh rke@manke03 lsblk
ssh rke@manke03 sudo reboot
ssh rke@manke04 sudo rm -rf /var/lib/rook
ssh rke@manke04 'ls /dev/mapper/ceph-* | xargs -I% -- echo /sbin/dmsetup remove %'
ssh rke@manke04 sudo rm -rf /dev/ceph-*
ssh rke@manke04 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300850"
ssh rke@manke04 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300850" bs=1M count=100 oflag=direct,dsync
ssh rke@manke04 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300850"
ssh rke@manke04 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301034"
ssh rke@manke04 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301034" bs=1M count=100 oflag=direct,dsync
ssh rke@manke04 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301034"
ssh rke@manke04 lsblk
ssh rke@manke04 sudo reboot
ssh rke@manke05 sudo rm -rf /var/lib/rook
ssh rke@manke05 'ls /dev/mapper/ceph-* | xargs -I% -- echo /sbin/dmsetup remove %'
ssh rke@manke05 sudo rm -rf /dev/ceph-*
ssh rke@manke05 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301063"
ssh rke@manke05 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301063" bs=1M count=100 oflag=direct,dsync
ssh rke@manke05 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301063"
ssh rke@manke05 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301344"
ssh rke@manke05 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301344" bs=1M count=100 oflag=direct,dsync
ssh rke@manke05 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301344"
ssh rke@manke05 lsblk
ssh rke@manke05 sudo reboot
ssh rke@manke06 sudo rm -rf /var/lib/rook
ssh rke@manke06 'ls /dev/mapper/ceph-* | xargs -I% -- echo /sbin/dmsetup remove %'
ssh rke@manke06 sudo rm -rf /dev/ceph-*
ssh rke@manke06 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301058"
ssh rke@manke06 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301058" bs=1M count=100 oflag=direct,dsync
ssh rke@manke06 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301058"
ssh rke@manke06 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300784"
ssh rke@manke06 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300784" bs=1M count=100 oflag=direct,dsync
ssh rke@manke06 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300784"
ssh rke@manke06 lsblk
ssh rke@manke06 sudo reboot
ssh rke@manke07 sudo rm -rf /var/lib/rook
ssh rke@manke07 'ls /dev/mapper/ceph-* | xargs -I% -- echo /sbin/dmsetup remove %'
ssh rke@manke07 sudo rm -rf /dev/ceph-*
ssh rke@manke07 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301068"
ssh rke@manke07 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301068" bs=1M count=100 oflag=direct,dsync
ssh rke@manke07 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301068"
ssh rke@manke07 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300787"
ssh rke@manke07 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300787" bs=1M count=100 oflag=direct,dsync
ssh rke@manke07 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T300787"
ssh rke@manke07 lsblk
ssh rke@manke07 sudo reboot
ssh rke@manke08 sudo rm -rf /var/lib/rook
ssh rke@manke08 'ls /dev/mapper/ceph-* | xargs -I% -- echo /sbin/dmsetup remove %'
ssh rke@manke08 sudo rm -rf /dev/ceph-*
ssh rke@manke08 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301067"
ssh rke@manke08 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301067" bs=1M count=100 oflag=direct,dsync
ssh rke@manke08 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301067"
ssh rke@manke08 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301313"
ssh rke@manke08 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301313" bs=1M count=100 oflag=direct,dsync
ssh rke@manke08 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301313"
ssh rke@manke08 lsblk
ssh rke@manke08 sudo reboot
ssh rke@manke09 sudo rm -rf /var/lib/rook
ssh rke@manke09 'ls /dev/mapper/ceph-* | xargs -I% -- echo /sbin/dmsetup remove %'
ssh rke@manke09 sudo rm -rf /dev/ceph-*
ssh rke@manke09 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301065"
ssh rke@manke09 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301065" bs=1M count=100 oflag=direct,dsync
ssh rke@manke09 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301065"
ssh rke@manke09 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301022"
ssh rke@manke09 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301022" bs=1M count=100 oflag=direct,dsync
ssh rke@manke09 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301022"
ssh rke@manke09 lsblk
ssh rke@manke09 sudo reboot
ssh rke@manke10 sudo rm -rf /var/lib/rook
ssh rke@manke10 'ls /dev/mapper/ceph-* | xargs -I% -- echo /sbin/dmsetup remove %'
ssh rke@manke10 sudo rm -rf /dev/ceph-*
ssh rke@manke10 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301061"
ssh rke@manke10 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301061" bs=1M count=100 oflag=direct,dsync
ssh rke@manke10 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301061"
ssh rke@manke10 sudo sgdisk --zap-all "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301031"
ssh rke@manke10 sudo dd if="/dev/zero" of="/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301031" bs=1M count=100 oflag=direct,dsync
ssh rke@manke10 sudo blockdev --rereadpt "/dev/disk/by-id/nvme-SAMSUNG_MZQL21T9HCJR-00A07_S64GNE0T301031"
ssh rke@manke10 lsblk
ssh rke@manke10 sudo reboot
```

File: manke/rook-ceph/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+complete ceph cluster tear down (will loose all data)
+---
+
+Run from rke user env:
+
+_Must be run prior to the node cleanup._
+
+```bash
+./scripts/rook-ceph-k8s-cleanup.sh
```
Comment (jhoblitt) [inline manke/rook-ceph/README.md:31]: This is no longer true because of `rook-remover`.  We should start cleaning up or remove the rook-ceph READMEs.

Comment (cbarria) [reply manke/rook-ceph/README.md:31]: yeah, you are right, i will remove de READMEs for now, probably will have to update them in the future

File: manke/rook-ceph/scripts/rook-ceph-node-cleanup.sh (hunk 2 above)
Comment (cbarria) [reply manke/rook-ceph/scripts/rook-ceph-node-cleanup.sh:8]: totally true, too much risk leaving a script that wipes every disk on one command laying around there =S

File: manke/rook-ceph/README.md (hunk 3 above)
Comment (jhoblitt) [reply manke/rook-ceph/README.md:31]: It is possible for incorrect docs to actually be a worse scenario than have no docs at all.

**Comment by jhoblitt** [comment]
@cbarria please squash the commits.

**Review by jhoblitt — APPROVED** [review]

**Pull request merged**